### PR TITLE
Ticket 39: cypress

### PR DESCRIPTION
**Pull request by maximFeldman1**
I add Cypree library and write some test

### CODE REVIEW

File: cypress/e2e/productPage.cy.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+describe("home page", () => {
+  it("create new product", () => {
+    cy.visit("http://127.0.0.1:3000/products");
```
Comment (itairom-appsforce) [inline cypress/e2e/productPage.cy.ts]:
```suggestion
    cy.visit(URL.PRODUCTS);
```

File: cypress/e2e/productPage.cy.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    cy.findByTestId("product-name__input").type("nike");
+    cy.findByTestId("product-brand__input").type("Nike");
+    cy.findByTestId("product-image__input").type(
+      "https://www.pexels.com/photo/close-up-portrait-of-lion-247502/"
+    );
+    cy.findByTestId("submit__button").click();
+  });
+
+  it("delete product", () => {
+    cy.visit("http://127.0.0.1:3000/products");
```
Comment (itairom-appsforce) [inline cypress/e2e/productPage.cy.ts]: same

File: cypress/support/commands.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+// declare global {
+//   namespace Cypress {
+//     interface Chainable {
+//       login(email: string, password: string): Chainable<void>
+//       drag(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
+//       dismiss(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
+//       visit(originalFn: CommandOriginalFn, url: string, options: Partial<VisitOptions>): Chainable<Element>
+//     }
+//   }
+// }
```
Comment (itairom-appsforce) [inline cypress/support/commands.ts]: remove comments

File: src/Pages/Products/components/ProductsTable/index.tsx [hunk 4]
```diff
@@ -17,7 +17,7 @@ const Root = styled.div`
 
 export const ProductsTable = () => {
   const { products, refetch } = useProductContext();
-  
+
```
Comment (itairom-appsforce) [inline src/Pages/Products/components/ProductsTable/index.tsx]: remove line

File: cypress/e2e/productPage.cy.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    cy.wait(1000);
+    cy.get(
+      "#root > div.Root-sc-ktahde-0.fVzYgt > div > div.Root-sc-qv9mpy-0.byYsWa.ag-theme-alpine > div > div > div.ag-root-wrapper-body.ag-focus-managed.ag-layout-normal > div.ag-root.ag-unselectable.ag-layout-normal > div.ag-body-viewport.ag-row-no-animation.ag-layout-normal > div.ag-center-cols-clipper > div > div > div.ag-row-odd.ag-row-no-focus.ag-row.ag-row-level-0.ag-row-position-absolute.ag-row-last > div:nth-child(6) > div > button"
+    ).click();
```
Comment (amitmarko) [inline cypress/e2e/productPage.cy.ts]: 😱 
Please use `data-testid` and don't selectors like that

File: cypress/e2e/productPage.cy.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    cy.findByText("Delete").click();
+    cy.findByTestId("yes__button").click();
+  });
+
+  it("doesn`t delete product", () => {
+    cy.visit("http://127.0.0.1:3000/products");
+
+    cy.wait(1000);
+    cy.get(
+      "#root > div.Root-sc-ktahde-0.fVzYgt > div > div.Root-sc-qv9mpy-0.byYsWa.ag-theme-alpine > div > div > div.ag-root-wrapper-body.ag-focus-managed.ag-layout-normal > div.ag-root.ag-unselectable.ag-layout-normal > div.ag-body-viewport.ag-row-no-animation.ag-layout-normal > div.ag-center-cols-clipper > div > div > div.ag-row-odd.ag-row-no-focus.ag-row.ag-row-level-0.ag-row-position-absolute.ag-row-last > div:nth-child(6) > div > button"
```
Comment (amitmarko) [inline cypress/e2e/productPage.cy.ts:27]: 😱
Please use data-testid and don't selectors like that

File: cypress/e2e/productPage.cy.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    ).click();
+    cy.findByText("Delete").click();
+    cy.findByTestId("yes__button").click();
+  });
+
+  it("doesn`t delete product", () => {
+    cy.visit("http://127.0.0.1:3000/products");
```
Comment (amitmarko) [inline cypress/e2e/productPage.cy.ts]:
```suggestion
    cy.visit(URL.PRODUCTS);
```

File: cypress/e2e/productPage.cy.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+    cy.wait(1000);
```
Comment (amitmarko) [inline cypress/e2e/productPage.cy.ts]:
```suggestion
```

File: cypress/e2e/productPage.cy.ts (hunk 1 above)
Comment (maximFeldman1) [reply cypress/e2e/productPage.cy.ts]: Fixed

File: cypress/e2e/productPage.cy.ts (hunk 2 above)
Comment (maximFeldman1) [reply cypress/e2e/productPage.cy.ts]: Fixed

File: cypress/support/commands.ts (hunk 3 above)
Comment (maximFeldman1) [reply cypress/support/commands.ts]: Fixed

File: cypress/e2e/productPage.cy.ts (hunk 7 above)
Comment (maximFeldman1) [reply cypress/e2e/productPage.cy.ts]: Fixed

File: cypress/e2e/productPage.cy.ts (hunk 8 above)
Comment (maximFeldman1) [reply cypress/e2e/productPage.cy.ts]: Fixed

**Pull request merged**